[Pua Li Xue] iP

pualixue opened the pull request:
# EightBit
> “Life is too complicated not to be orderly.” – Martha Stewart ([source](https://dansilvestre.com/productivity-quotes/#10_Life_is_too_complicated_not_to_be_orderly_-_Martha_Stewart))

EightBit frees your mind of having to remember things you need to do. It's
* text-based
* easy to learn
* ~~fast~~ *SUPER* FAST to use

All you need to do is:
1. download it from [here](https://github.com/pualixue/ip/releases/tag/v0.1)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😄 

And it is **FREE**!

Features:
- [x] List your tasks
- [x] Add/remove tasks
- [x] Mark tasks as done
- [x] Search for tasks
- [ ] Reminders (coming soon)

#
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    EightBit eightBit = new EightBit(FILEPATH);
    eightBit.run();
}
```

Commit:
Add Gradle support

pualixue committed:
Greet user and echo commands

pualixue committed:
Store and list user's input text

pualixue committed:
Allow marking tasks as done

pualixue committed:
Allow tracking todos, deadlines, events

pualixue committed:
Test using I/O redirection technique

pualixue committed:
Handle errors

pualixue committed:
Add support for deleting tasks

pualixue committed:
Use Enums to represent command types, add Javadoc comments

pualixue committed:
Save tasks to hard disk

pualixue committed:
Store dates and times

pualixue committed:
Merge branch 'branch-Level-7'

pualixue committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/EightBit.java

pualixue committed:
Refactor code to extract out closely related code as classes

pualixue committed:
Organize classes into packages

pualixue committed:
Add JUnit tests

pualixue committed:
Package the app as a JAR file

pualixue committed:
Add JavaDoc comments

pualixue committed:
Tweak the code to comply with coding standard

pualixue committed:
Allows finding tasks by searching for keyword

pualixue committed:
Merge branch 'branch-A-JavaDoc'

pualixue committed:
Merge branch 'branch-A-CodingStandard'

pualixue committed:
Merge branch 'branch-Level-9'

pualixue committed:
Add more Javadoc comments

pbthang commented on src/main/java/eightbit/EightBitException.java:
```diff
@@ -0,0 +1,25 @@
+     * @param errorMessage Error message to be shown to user.
+     */
+    public EightBitException(String errorMessage) {
+        this.errorMessage = errorMessage;
```
I think you should use the constructor of the parent class.
```suggestion
        super(errorMessage);
```

pbthang commented on src/main/java/eightbit/task/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+    @Override
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm");
+        return "[D]" + super.toString() + " (by: " + by.format(formatter) + ")";
```
In this case, should we use `String.format()` instead, since for me, it's more readable and less verbose?

pbthang commented on src/main/java/eightbit/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+/**
+ * Represents a task that can be completed.
+ */
+public class Task {
```
Should this class be abstract, because it's only instantiated in the forms of Todo, Event, etc?
```suggestion
public abstract class Task {
```

pualixue replied:
Thanks for your suggestions!

pbthang reviewed (commented):
Looks good. Some minor changes requested.

pualixue committed:
Merge branch 'add-gradle-support' of https://github.com/pualixue/ip into branch-A-Gradle

pualixue committed:
Set up Gradle to build and run EightBit

pualixue committed:
Merge branch 'branch-A-Gradle'

pualixue committed:
Use CheckStyle to detect coding style violations

pualixue committed:
Merge branch 'branch-A-CheckStyle'

pualixue committed:
Add a GUI to EightBit

pualixue committed:
Merge branch 'branch-Level-10'

chongjunwei commented on src/main/java/eightbit/command/FindCommand.java:
```diff
@@ -0,0 +1,58 @@
+        ArrayList<Task> results = new ArrayList<>();
+        for (int i = 0; i < taskList.size(); i++) {
+            Task task = taskList.get(i);
+            if (task.getDescription().toLowerCase(Locale.ROOT).contains(keyword)) { // case-insensitive search
```
The comment should be indented relative to its position in the code, in this case in line with the if (task.getDescription()...) statement.

chongjunwei commented on src/main/java/eightbit/command/EventCommand.java:
```diff
@@ -0,0 +1,37 @@
+     * @return The response after executing the command.
+     */
+    @Override
+    public String execute(TaskList taskList, Storage storage) {
```
I think it would be better to name the method after the functionality, instead of execute() which may not be the most intuitive. Perhaps something like addEventToList()?

chongjunwei reviewed (commented):
Looks good, just a few minor comments based on the code quality and standard guidelines given. Good job!

pualixue committed:
Create JAR file

pualixue committed:
Add assertions

There are some assumptions that should hold at various points of code.
Using assertions ensures the program executes correctly.

Let's use the `assert` feature to document the important assumptions.

pualixue committed:
Improve code quality

There are some complicated expressions and magic numbers. These can
affect the readability of the code.

Let's refactor the code to adhere to coding standards and quality.

pualixue committed:
Merge pull request #1 from pualixue/branch-A-Assertions

Add assertions

pualixue committed:
Merge branch 'master' into branch-A-CodeQuality

pualixue committed:
Merge pull request #2 from pualixue/branch-A-CodeQuality

Improve code quality

pualixue committed:
Use GitHub Actions to set up Continuous Integration (CI)

pualixue committed:
Provide a way to tag tasks

Users can only add all the information about a task in its description.

Let's allow users to add multiple tags to tasks.

Tags allow users to specify additional information about a task and
differentiate them more easily.

pualixue committed:
Merge branch 'branch-C-Tagging'

pualixue committed:
Improve the GUI

pualixue committed:
Merge branch 'branch-A-BetterGui'

pualixue committed:
Add test cases for different Task

pualixue committed:
Set theme jekyll-theme-cayman

pualixue committed:
Add a User Guide (UG)

pualixue committed:
Merge branch 'master' of https://github.com/pualixue/ip

pualixue committed:
Fix minor issue in parsing commands and JavaFX version

pualixue committed:
docs/README.md: Update download link
